Source/CBL_Router.m

```diff
@@ -573,6 +571,9 @@ - (void) processRequestRanges {
         return;
     }
 
+    if (to < from)
```

Comment:
I think the `to < from` check needs to happen before the `from >= bodyLength` check. For example, if the bodyLength is 100 and the range is `500-400`, the response status should be 200 (since the range is invalid) not 416.